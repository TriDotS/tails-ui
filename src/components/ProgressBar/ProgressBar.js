import PropTypes from 'prop-types'
import React from 'react'

const ProgressBar = ({ color, percent }) => (
  <div className="w-full">
    <div className="shadow w-full bg-grey-light rounded">
      <div
        className={`bg-${color} rounded text-xs leading-none py-1 text-center text-white`}
        style={{ width: percent }}
      >
        {percent}
      </div>
    </div>
  </div>
)

ProgressBar.defaultProps = {
  color: 'teal'
}

ProgressBar.propTypes = {
  color: PropTypes.string,
  percent: PropTypes.string.isRequired
}

export default ProgressBar
