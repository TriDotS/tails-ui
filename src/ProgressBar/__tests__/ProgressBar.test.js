import ProgressBar from '../ProgressBar'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<ProgressBar />, div)
})

test('progress bar snapshot', () => {
  const tree = renderer.create(<ProgressBar />).toJSON()
  expect(tree).toMatchSnapshot()
})
