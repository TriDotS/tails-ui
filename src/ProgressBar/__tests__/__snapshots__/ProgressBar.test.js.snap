// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress bar snapshot 1`] = `
<div
  className="w-full"
>
  <div
    className="shadow w-full bg-grey-light rounded"
  >
    <div
      className="bg-teal rounded text-xs leading-none py-1 text-center text-white"
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;
